src/vs/editor/contrib/stickyScroll/browser/stickyScrollWidget.ts

```diff
@@ -21,6 +21,12 @@ export class StickyScrollWidgetState {
 		readonly lineNumbers: number[],
 		readonly lastLineRelativePosition: number
 	) { }
+
+	public equals(other: StickyScrollWidgetState | undefined): boolean {
+		return !!other && this.lastLineRelativePosition === other.lastLineRelativePosition
+			&& this.lineNumbers.length === other.lineNumbers.length
```

Comment:
suggestion to use [`arrays.ts#equals`](https://github.com/microsoft/vscode/blob/b529a3be0bc07437248c6578606e94ab3cc02731/src/vs/base/common/arrays.ts#L27)